optical/converter/sagemaker.py

```diff
@@ -4,14 +4,87 @@
 Created: Wednesday, 31st March 2021
 """
 
-from .base import FormatSpec
-from typing import Union
+import json
 import os
+from pathlib import Path
+from typing import Union
+import warnings
+import pandas as pd
+from tqdm.auto import tqdm
+
+from .base import FormatSpec
+from .utils import exists, find_job_metadata_key, get_annotation_dir, get_image_dir
 
 
 class SageMaker(FormatSpec):
+    """Class to handle sagemaker '.manifest' annotation transformations
+
+    Args:
+        FormatSpec : Base class to inherit class attributes
+    """
+
     def __init__(self, root: Union[str, os.PathLike]):
-        pass
+        self.root = root
+        self._image_dir = get_image_dir(root)
+        self._annotation_dir = get_annotation_dir(root)
+        self._has_image_split = False
+        assert exists(self._image_dir), "root is missing `images` directory."
+        assert exists(self._annotation_dir), "root is missing `annotations` directory."
+        self._splits = self._find_splits()
+        self._resolve_dataframe()
+
+    def _find_splits(self):
+        im_splits = [x.name for x in Path(self._image_dir).iterdir() if x.is_dir()]
+        ann_splits = [x.stem for x in Path(self._annotation_dir).glob("*.manifest")]
+
+        if im_splits:
+            self._has_image_split = True
+
+        if not len(ann_splits):
+            warnings.warn(f"no annotation file found inside {self._annotation_dir}")
+
+        no_anns = set(im_splits).difference(ann_splits)
+        if no_anns:
+            warnings.warn(f"no annotation found for {', '.join(list(no_anns))}")
+        return ann_splits
 
     def _resolve_dataframe(self):
-        return super()._resolve_dataframe()
+        master_data = {
```

Comment:
@bishwarup307 , I see `class_id` in coco `master_df`, is it supposed to be followed by all other formats `master_df`s? `visualizer` currently requires it as a mandatory column, let me if need change or all `master_df` will have `class_id` column.

Reply:
`class_id` is a mandatory field for for the `master_df`